lib/driver.js

```diff
@@ -467,12 +467,12 @@ export class XCUITestDriver extends BaseDriver {
       // ensure WDA gets our defaults instead of whatever its own might be
       await this.updateSettings(wdaSettings);
 
-      // turn on mjpeg stream reading if requested
-      if (this.opts.mjpegScreenshotUrl) {
-        this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
-        this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
-        await this.mjpegStream.start();
+      if (this.isRealDevice() && this.opts.mjpegServerPort) {
+        await this.allocateMjpegServerPort(caps);
```

Comment:
I expect this behaviour to be aligned with the screen recorder defined in the screenrecord.js 
It also tries to create a port forward when a new recording is started, so the logic there must not clash with the existing port forwarding rule

Reply:
@mykola-mokhnach It does not, coz in the screenrecord.js it catchs and silently gives an log. So I didn't bother to check anything there 

https://github.com/appium/appium-xcuitest-driver/blob/master/lib/commands/recordscreen.js#L69-L72

Reply:
This must be updated, because even though we ignore the warning here, the port forwarding rule will be removed when the screen recording is finished. Possible bug scenario:
- start the test with port number given in caps and mjpeg server
- while the test is running start a screen recording
- stop the screen recording

Actual result:
port forwarding is killed, mjpeg server is down

Expected result:
port forwarding must still be alive

As a consequence we now must rely on the fact that if mjpeg port was provided as a capability then the port forward lifecycle is controlled by the driver instead of the screen recording APIs

Reply:
![giphy](https://github.com/user-attachments/assets/e52b7baf-4ec3-40f9-9ffc-0e7cdb045ecc)